src/cqrs/Aggregate.ts

```diff
@@ -81,4 +103,37 @@ export class Aggregate {
       this.aggregateRoles.set(entityId, newRoles);
     }
   }
+
+  /**
+   * Increate events counter by 1
+   *
+   * @param event
+   * @returns number
+   */
+  applyEvent(eventId: string): Aggregate {
+    // if eventId has not been applied.
+    if (!this.eventApplied(eventId)) {
```

Comment:
We'll need a sort of read-repair process here as well... if the event has no ordinal we should give it one based on the order in which it arrives.